content/courses/advanced-ada/chapters/types.rst

```diff
@@ -445,10 +445,123 @@ Mutually dependent types
 Type view
 ---------
 
-.. todo::
+Ada distinguishes between the partial and the full view of a type. The full
+view is a type declaration that contains all the information needed by the
+compiler. For example, the following declaration of type :ada:`R` represents
+the full view of this type:
 
-    Complete section!
+.. code:: ada compile_button project=Courses.Advanced_Ada.Types.Full_View
+
+    package Full_View is
+
+       --  Full view of the R type:
+       type R is record
+          I : Integer;
+       end record;
+
+    end Full_View;
+
+As soon as we start applying encapsulation |mdash| via the :ada:`private`
+keyword |mdash| to a specific type, we need to split the type declaration into
+a partial view and a full view. For example:
```

Comment:
Maybe something like:
"a specific type, we are introducing a partial view and making only that view compile-time visible to clients. Doing so requires us to introduce the private part of the package (unless already present)."

Reply:
Yes, I like it.